src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,41 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder salaryInfo = new StringBuilder();
+        int salary;
+        LocalDate start = LocalDate.parse(dateFrom, formatter);
+        LocalDate end = LocalDate.parse(dateTo, formatter);
+        salaryInfo.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String name : names) {
+            salary = 0;
+            for (String datum : data) {
+                String[] lines = datum.split(" ");
+                LocalDate currentDate = LocalDate.parse(lines[0], formatter);
+                String nameUser = lines[1];
+                String workHours = lines[2];
+                String incomePerHour = lines[3];
+                if (!currentDate.isBefore(start) && !currentDate.isAfter(end)
+                        && name.equals(nameUser)) {
+                    salary += Integer.parseInt(workHours) * Integer.parseInt(incomePerHour);
+                } else if (currentDate.isEqual(start) && name.equals(nameUser)
+                        || currentDate.isEqual(end) && name.equals(nameUser)) {
+                    salary += Integer.parseInt(workHours) * Integer.parseInt(incomePerHour);
```

Comment:
The condition checking for equality with `start` and `end` dates is redundant. The existing condition on lines 26-28 already includes these dates. You can remove this block to simplify the logic.